allennlp/predictors/predictor.py

```diff
@@ -40,7 +40,9 @@ class Predictor(Registrable):
     that can be used for serving models through the web API or making predictions in bulk.
     """
 
-    def __init__(self, model: Model, dataset_reader: DatasetReader) -> None:
+    def __init__(self, model: Model, dataset_reader: DatasetReader, frozen: bool = True) -> None:
+        if frozen:
+            model = model.eval()
```

Comment:
Just call `model.eval()` here, not `model = model.eval()`.  That's what you do below.